packages/spartan/extra/gen_resolvconf.py

```diff
@@ -4,32 +4,64 @@
 import os
 import random
 import socket
+import subprocess
 import sys
 
+from ctypes import cdll
 import dns.query
 
 
+BASE_CONTENTS = """# Generated by gen_resolvconf.py. Do not edit.
+# Change configuration options by changing DC/OS cluster configuration.
+# This file must be overwritten regularly for proper cluster operation around
+# master failure.
+
+options timeout:1
+options attempts:3
+
+"""
+
 # Constants
 MAX_SERVER_COUNT = 3
 SPARTANS = ['198.51.100.1', '198.51.100.2', '198.51.100.3']
 
+# We assume that RESOLVCONF_DIR_PATH is precreated
+RESOLVCONF_DIR_PATH = '/var/lib/dcos/resolvconf'
+RESOLVCONF_PATH = os.path.join(RESOLVCONF_DIR_PATH, 'resolv.conf')
+DCOS_BIND_MOUNT_POINT = '/tmp/dcos'
```

Comment:
I think this is dangerous -- some tmpwatch thing is going to roll through and clean this up.

/run/dcos is probably a much better place for this file.